### PR TITLE
[Layers] Add qwenRope support for Qwen1.0 in CB mode

### PR DESCRIPTION
**Pull request by abenmao**

### CODE REVIEW

File: src/kernels/rotary_embedding_kernels.cpp [hunk 1]
```diff
@@ -450,6 +449,81 @@ void qwenApplyRotaryPosEmbeding(float16_t *query, float16_t *key, int qStride, i
...
+template <typename T>
+static inline void qwenApplyRotaryPosEmbed(T *query, T *key, float *emb_cos, float *emb_sin, int qStride, int kStride,
+        int dim, const float *logn, int maxSupportedSeqLength, int totSeqLen, int qHeads, int kHeads,
+        const int *positionIds) {
+    const int half = (dim + 1) / 2;
+    const int heads = std::max(qHeads, kHeads);
+
+#pragma omp parallel for collapse(2)
+    for (int head = 0; head < heads; ++head) {
+        for (int seq = 0; seq < totSeqLen; ++seq) {
```
Comment (pujiang2018) [inline src/kernels/rotary_embedding_kernels.cpp:461]: For next step, considering first token, should we swap the 2 loops to make each thread accessing contiguous memory? may deserve to test such implementation.
OK for current version.

**Review by pujiang2018 — APPROVED** [review]

**Comment by changqi1** [comment]
I think this kernel APIs on continuous batching version and on continuous batching version are the same. Next step. we could merge two into one kernel API.

**Comment by abenmao** [comment]
> I think this kernel APIs on continuous batching version and on continuous batching version are the same. Next step. we could merge two into one kernel API. OK for current version.

Yes, maybe we can remove the older version in the next step.

File: src/kernels/rotary_embedding_kernels.cpp (hunk 1 above)
Comment (abenmao) [reply src/kernels/rotary_embedding_kernels.cpp:461]: Got it~

**Pull request merged**